### PR TITLE
feat(ui): handle cluster/pod 404s and redirects when deleting

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Update the cluster and pod detail views to display a not-found message if the id is invalid.
- Redirect from the delete form when the cluster/pod is deleted.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit /MAAS/r/kvm/virsh/9999.
- You should see a not-found message.
- Visit /MAAS/r/kvm/lxd/9999.
- You should see a not-found message.
- Visit /MAAS/r/kvm/lxd/cluster/9999.
- You should see a not-found message.
- Go to the /KVM then to the Virsh tab and create a new virsh pod.
- Go to the new Virsh and use the take action menu to open the delete form.
- Click "Remove KVM" and you should get redirected back to the list and there should be a message the that KVM was removed.

## Fixes

Fixes: canonical-web-and-design/maas-ui#3149.

### CODE REVIEW

File: ui/src/app/kvm/components/KVMHeaderForms/DeleteForm/DeleteForm.tsx [hunk 1]
```diff
@@ -90,7 +101,17 @@ const DeleteForm = ({
...
+        dispatch(
+          messageActions.add(
+            `${pod ? "KVM" : "Cluster"} removed successfully`,
+            NotificationSeverity.INFORMATION
+          )
+        );
+        history.push({ pathname: kvmURLs.kvm });
+      }}
       processingCount={deletingCount}
+      saved={deleted}
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/components/KVMHeaderForms/DeleteForm/DeleteForm.tsx:114]: We should also make sure the delete action was successful, i.e. there are no errors

**Review by Caleb-Ellis — APPROVED** [review]
LGTM, just one comment on deleting

File: ui/src/app/kvm/components/KVMHeaderForms/DeleteForm/DeleteForm.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/kvm/components/KVMHeaderForms/DeleteForm/DeleteForm.tsx:126]: I've added this (and a bunch of tests). Would you mind taking another look? The changes are in the second commit.

**Comment by Caleb-Ellis** [comment]
@huwshimi yep looks good!

**Pull request merged**